pallets/portfolio/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+            Option<PortfolioName>;
+        /// Asset balances of portfolios.
+        pub PortfolioAssetBalances get(fn portfolio_asset_balances):
+            double_map hasher(blake2_128_concat) PortfolioId, hasher(blake2_128_concat) Ticker =>
+            T::Balance;
+        /// The next portfolio sequence number of an identity.
+        pub NextPortfolioNumber get(fn next_portfolio_number):
+            map hasher(blake2_128_concat) IdentityId => PortfolioNumber;
+    }
+}
+
+decl_event! {
+    pub enum Event<T> where
+        Balance = <T as CommonTrait>::Balance,
+    {
+        /// The portfolio has been successfully created.
+        PortfolioCreated(IdentityId, PortfolioNumber, PortfolioName),
+        /// The portfolio has been successfully removed.
+        PortfolioDeleted(IdentityId, PortfolioNumber),
+        /// A token amount has been moved from one portfolio to another. `None` denotes the default
+        /// portfolio of the DID.
+        MovedBetweenPortfolios(
+            IdentityId,
+            Option<PortfolioNumber>,
+            Option<PortfolioNumber>,
+            Ticker,
+            Balance
+        ),
+        /// The portfolio identified with `num` has been renamed to `name`.
+        PortfolioRenamed(IdentityId, PortfolioNumber, PortfolioName),
+        /// All non-default portfolio numbers and names of a DID.
+        UserPortfolios(IdentityId, Vec<(PortfolioNumber, PortfolioName)>),
+    }
+}
+
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+        /// The portfolio doesn't exist.
+        PortfolioDoesNotExist,
+        /// Insufficient balance for a transaction.
+        InsufficientPortfolioBalance,
+        /// The source and destination portfolios should be different.
+        DestinationIsSamePortfolio,
+        /// The portfolio couldn't be renamed because the chosen name is already in use.
+        PortfolioNameAlreadyInUse,
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        /// The event logger.
+        fn deposit_event() = default;
+
+        /// Creates a portfolio.
+        #[weight = 200_000]
+        pub fn create_portfolio(origin, name: PortfolioName) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            let name_uniq = <Portfolios>::iter_prefix(&did).all(|n| n.1 != name);
+            ensure!(name_uniq, Error::<T>::PortfolioNameAlreadyInUse);
+            let num = Self::get_next_portfolio_number(&did);
+            <Portfolios>::insert(&did, &num, name.clone());
+            Self::deposit_event(RawEvent::PortfolioCreated(did, num, name));
+            Ok(())
+        }
+
+        /// Deletes a user portfolio and moves all its assets to the default portfolio.
+        #[weight = 1_000_000]
+        pub fn delete_portfolio(origin, num: PortfolioNumber) -> DispatchResult {
```

Comment:
Will add comments. My take on vertical spaces inside functions is that they are ellipsed comments.

Reply:
Ellipsed comments? (The reason I suggested vertical spaces is that otherwise the code becomes overly dense and there's no "breathing" if you catch my drift.)

Reply:
Yes, when people make a vertical space, that space carries a meaning (like ellipsis does in natural language) that is not explained with a comment. That is what I mean by an ellipsed comment. It's better to give a comment instead if you know why there should be a pause here. Ellipsis is good in fiction and less so in technical literature and even less so in code.

Reply:
I see; I use vertical spaces as a grouping mechanism and to reduce clutter:

```rust
// Foo section of code.
foo_step1();
foo_step2();

// Bar section of code.
bar_step1();
bar_step2();
```

This makes it clearer that the comment above is a section comment, as opposed to a comment regarding just `*_step1();`. You could write out this fact in the comment, but that seems to me overly verbose.
Anyways, to each their own, and if you prefer it this way in your code then go for it. :)

Reply:
Are you introducing a new convention about vertical space before a comment? :)

BTW comments can be coloured less brightly and act as a separator.

Reply:
Hehe; That's always how I've done it & seen it done in e.g., rust-lang/rust; it works pretty well, but it's only a convention if others agree. 😉